### PR TITLE
feat: adds support for max locks and unlocking

### PR DESCRIPTION
**Pull request by kobigurk**
Also fixes a lint issue.

### CODE REVIEW

File: coordinator-service/src/disk-coordinator.ts [hunk 1]
```diff
@@ -135,6 +139,22 @@ export class DiskCoordinator implements Coordinator {
...
+
+        const chunk = DiskCoordinator._getChunk(ceremony, chunkId)
+        if (chunk.lockHolder !== participantId) {
+            throw new Error(
+                `${participantId} does not hold lock on chunk ${chunkId}`,
+            )
+        }
+
+        chunk.lockHolder = null
+        chunk.metadata.lockHolderTime = null
```
Comment (silasbw) [inline coordinator-service/src/disk-coordinator.ts:153]: the original semantics of this field are the last time `lockHolder` is updated.  Analogous to mtime on a file.  The other "time" fields are the same (when the corresponding field is update, a new timestamp is recorded).

**Review by silasbw — COMMENTED** [review]
Did you mean to update the submodule stuff?

File: coordinator-service/src/disk-coordinator.ts (hunk 1 above)
Comment (kobigurk) [reply coordinator-service/src/disk-coordinator.ts:153]: That's a good point. I'll change it to keep the semantics.

**Comment by kobigurk** [comment]
> Did you mean to update the submodule stuff?

Yep. I'm constantly improving it right now.

**Comment by kobigurk** [comment]
@silasbw @mrsmkl do you think that the system would be more DoS resistant if we remove the unlocking logic and instead require the client work on the chunks they already have locked?

**Review by silasbw — APPROVED** [review]

**Comment by silasbw** [comment]
i'm not sure.  if the model is "participants might intentionally try to DoS", it seems like there are potentially many ways to DoS.

If the model is "participants might unintentionally DoS (e.g., via a client bug)", then what you purpose could close one possible vulnerability.  it doesn't seem substantial though and if the unlocking adds some usability / convenience, it seems work it.

**Comment by kobigurk** [comment]
> i'm not sure. if the model is "participants might intentionally try to DoS", it seems like there are potentially many ways to DoS.
> 
> If the model is "participants might unintentionally DoS (e.g., via a client bug)", then what you purpose could close one possible vulnerability. it doesn't seem substantial though and if the unlocking adds some usability / convenience, it seems work it.

I'm convinced. Let's not do it!

**Pull request merged**